src/shogun/distance/EuclideanDistance.h

```diff
@@ -125,6 +161,21 @@ class CEuclideanDistance: public CRealDistance
 	protected:
 		/** if application of sqrt on matrix computation is disabled */
 		bool disable_sqrt;
+
+		/** if squared norms of rhs features are used */
+		bool m_rhs_sq_norms;
+
+		/** if squared norms of lhs features are used */
+		bool m_lhs_sq_norms;
+
+		/** i dot products are used in distance computations */
+		bool m_dot_enabled;
+
+		/** squared norms from features of right hand side to be used if dot enabled */
+		SGVector<float64_t> m_rhs_squared_norms;
```

Comment:
I see your point here.
In fact, I think it does not make sense to precompute the rhs squared norms at all. Distance is only computed once no?
Maybe only precompute lhs?